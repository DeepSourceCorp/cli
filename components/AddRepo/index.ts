import AddRepoModal from './AddRepoModal.vue'
import RepoCard from './RepoCard.vue'
import SyncRepoAlert from './SyncRepoAlert.vue'

export { AddRepoModal, RepoCard, SyncRepoAlert }
