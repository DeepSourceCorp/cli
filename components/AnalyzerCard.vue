<template>
  <div class="relative h-72 min-h-1/2 md:h-40 bg-ink-400 p-2 rounded-sm flex flex-col space-y-3 md:min-w-1/2 md:w-1/2 box-border overflow-scroll"
        :class="{ 'justify-center items-center space-y-2': configList.length }">
    <!-- Header -->
    <div v-if="configList.length > 0" class="flex w-full items-center space-x-2">
        <z-icon :icon="icon" size="small" color="robin"></z-icon>
        <!-- TODO: Get Repo name and handle name -->
        <div class="flex-1 text-sm text-vanilla-200 font-bold">
            {{ name }}
        </div>
        <div class="cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
    </div>
    <!-- Config Section -->
    <template v-for="config in configList">
        <div :key="config.title" class="flex flex-col space-y-2">
            <div class="text-xs tracking-wide text-slate uppercase">
                {{config.title}}
            </div>
            <div v-if="config.type == 'boolean'">
                <z-radio-group v-model="config.selected" class="flex space-x-4 text-xs">
                    <z-radio v-for="option in config.values" 
                            :key="option.value"
                            :value="option.value"
                            :label="option.label"
                            ></z-radio>
                </z-radio-group>
            </div>
            <div v-else-if="config.type == 'input'">
                <z-input v-model="config.selected" 
                        textSize="xs">
                </z-input>
            </div>
            <div v-else-if="config.type == 'list'" class="h-8">
                <!-- TODO: Customizable placeholder -->
                <z-select 
                    v-model="config.selected" 
                    class="z-10" 
                    placeholder="Choose a framework">
                    <z-option
                        v-for="item in config.values"
                        :key="item.value"
                        :label="item.label"
                        :value="item.value">
                        <div class="flex items-center">
                            <z-icon icon="activity" size="small" class="-mt-1 mr-2"></z-icon>
                            <span class="flex-1">{{item.label}}</span>
                        </div>
                    </z-option>
                </z-select>
            </div>
        </div>
    </template>
    <!-- No Config Data Section -->
    <template v-if="configList.length == 0">
        <div class="absolute top-2 right-2 cursor-pointer" @click="handleClick">
            <z-icon icon="x"
                size="small" 
                color="vanilla-200">
            </z-icon>
        </div>
        <z-icon :icon="icon" color="pink"></z-icon>
        <div class="font-bold text-vanilla-100 text-sm">{{name}}</div>
    </template>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon, ZInput, ZSelect, ZOption, ZRadioGroup, ZRadio } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon,
       ZInput,
       ZSelect,
       ZOption,
       ZRadioGroup,
       ZRadio
    }
})
export default class AnalyzerCard extends Vue {
    @Prop({default: 'lock'})
    icon!: string
    @Prop({default: ''})
    name!: string
    @Prop({default: []})
    configList!: Array<Record<string, unknown>>
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
