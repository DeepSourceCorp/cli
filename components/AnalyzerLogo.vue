<template>
  <div>
    <img
      v-if="analyzerLogo"
      v-tooltip="hideTooltip ? '' : name"
      class="w-auto flex-shrink-0"
      :class="sizeStyle"
      :src="analyzerLogo"
      :alt="name"
    />
    <z-icon
      v-else
      v-tooltip="hideTooltip ? '' : name"
      :icon="shortcode"
      color="transparent"
      :size="size"
    />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component({})
export default class AnalyzerLogo extends Vue {
  @Prop({ default: '' })
  analyzerLogo!: string

  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  shortcode!: string

  @Prop({ default: 'base' })
  size!: string

  @Prop({ default: false })
  hideTooltip: boolean

  get sizeStyle() {
    const sizes: Record<string, string> = {
      'x-small': 'w-3 h-3',
      small: 'w-4 h-4',
      base: 'w-5 h-5',
      medium: 'w-6 h-6',
      large: 'w-8 h-8'
    }

    return sizes[this.size] || 'w-5 h-4'
  }
}
</script>
