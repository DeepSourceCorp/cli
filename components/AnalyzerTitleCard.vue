<template>
  <div class="px-2 py-2 bg-ink-100 rounded-sm bg-ink-400 flex items-center space-x-2 cursor-pointer"
        @click="handleClick()">
    <z-icon :icon="icon" size="small" color="honey"></z-icon>
    <div class="flex-1 text-sm text-vanilla-200 font-bold">{{name}}</div>
    <z-icon icon="plus" size="small" class="cursor-pointer" color="vanilla-200"></z-icon>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import { ZIcon } from '@deepsourcelabs/zeal';
@Component({
    components: {
       ZIcon 
    }
})
export default class AnalyzerTitleCard extends Vue {
    @Prop({default: 'globe'})
    icon!: string
    @Prop({default: ''})
    name!: string
    public handleClick(event: Event): void {
        this.$emit('click', event)
    }
}
</script>
