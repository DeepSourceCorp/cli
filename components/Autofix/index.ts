import InfoIcon from './InfoIcon.vue'
import Info from './Info.vue'
import AutofixCard from './AutofixCard.vue'
import AutofixListItem from './AutofixListItem.vue'
import AutofixCodeDiff from './AutofixCodeDiff.vue'

export { InfoIcon, AutofixCard, AutofixListItem, AutofixCodeDiff, Info }
