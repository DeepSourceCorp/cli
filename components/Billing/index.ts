import PlanInfo from './PlanInfo.vue'
import BillingInfo from './BillingInfo.vue'
import InvoiceList from './InvoiceList.vue'
import UpdateBillingDetailsModal from './Modals/UpdateBillingDetailsModal.vue'

export { PlanInfo, BillingInfo, InvoiceList, UpdateBillingDetailsModal }
