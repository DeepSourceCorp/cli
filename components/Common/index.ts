import SidebarMenu from './SidebarMenu.vue'

export { SidebarMenu }
