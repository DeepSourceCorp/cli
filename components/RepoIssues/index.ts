import IssueActions from './IssueActions.vue'
import IssueDescription from './IssueDescription.vue'
import IssueDetailsHeader from './IssueDetailsHeader.vue'
import IssueEditor from './IssueEditor.vue'
import IssueOccurrenceSection from './IssueOccurrenceSection.vue'
import IssueList from './IssueList.vue'
import IgnoreIssueFilePattern from './Modals/IgnoreIssueFilePattern.vue'
import IgnoreIssueAllFiles from './Modals/IgnoreIssueAllFiles.vue'
import IgnoreIssueTestFiles from './Modals/IgnoreIssueTestFiles.vue'
import IgnoreIssueIntentional from './Modals/IgnoreIssueIntentional.vue'
import IgnoreIssueFalsePositive from './Modals/IgnoreIssueFalsePositive.vue'
import IgnoreIssueOccurrence from './Modals/IgnoreIssueOccurrence.vue'
import AutofixFileChooser from './Modals/AutofixFileChooser.vue'
import AutofixIssuesChooser from './Modals/AutofixIssuesChooser.vue'

export {
  IssueActions,
  IssueList,
  IssueDescription,
  IssueDetailsHeader,
  IssueEditor,
  IssueOccurrenceSection,
  IgnoreIssueFilePattern,
  IgnoreIssueAllFiles,
  IgnoreIssueTestFiles,
  IgnoreIssueIntentional,
  IgnoreIssueFalsePositive,
  IgnoreIssueOccurrence,
  AutofixFileChooser,
  AutofixIssuesChooser
}
