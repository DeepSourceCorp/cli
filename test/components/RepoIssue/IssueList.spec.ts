import { render } from '@testing-library/vue'
import { IssueList } from '~/components/RepoIssues'

describe('[[ IssueList ]]', () => {
  const baseProps = {
    totalCount: 3,
    pageSize: 25,
    description:
      '<p>Do not use a mutable like <code>list</code> or <code>dictionary</code> as a default value to an argument. Pythonâ€™s default arguments are evaluated once when the function is defined. Using a mutable default argument and mutating it will mutate that object for all future calls to the function as well.</p>\n\n<!--more-->\n\n<h3>Bad practice</h3>\n\n<div class="highlight markdown-rendered">\n<pre><span></span><code><span class="k">def</span> <span class="nf">my_function</span><span class="p">(</span><span class="n">elem</span><span class="p">,</span> <span class="n">l</span><span class="o">=</span><span class="p">[]):</span>\n    <span class="n">l</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">elem</span><span class="p">)</span>\n    <span class="k">return</span> <span class="n">l</span>\n\n<span class="nb">print</span><span class="p">(</span><span class="n">my_function</span><span class="p">(</span><span class="mi">2</span><span class="p">))</span> <span class="c1"># [2]</span>\n<span class="nb">print</span><span class="p">(</span><span class="n">my_function</span><span class="p">(</span><span class="mi">5</span><span class="p">))</span> <span class="c1"># [2, 5]</span>\n</code></pre>\n</div>\n\n<h2>Recommended:</h2>\n\n<div class="highlight markdown-rendered">\n<pre><span></span><code><span class="k">def</span> <span class="nf">my_function</span><span class="p">(</span><span class="n">elem</span><span class="p">,</span> <span class="n">l</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>\n    <span class="k">if</span> <span class="n">l</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>\n        <span class="n">l</span> <span class="o">=</span> <span class="p">[]</span>\n    <span class="n">l</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">elem</span><span class="p">)</span>\n    <span class="k">return</span> <span class="n">l</span>\n\n<span class="nb">print</span><span class="p">(</span><span class="n">my_function</span><span class="p">(</span><span class="mi">2</span><span class="p">))</span> <span class="c1"># [2]</span>\n<span class="nb">print</span><span class="p">(</span><span class="n">my_function</span><span class="p">(</span><span class="mi">5</span><span class="p">))</span> <span class="c1"># [5]</span>\n</code></pre>\n</div>\n',
    blobUrlRoot:
      'https://github.com/deepsourcelabs/demo-python/blob/2316e689facc8b4ea67258230b5cbce5c7d6c8d4/',
    checkId: 'Q2hlY2s6enFkbmpv',
    canIgnoreIssues: true,
    issueOccurrences: [
      {
        id: 'Q2hlY2tJc3N1ZTpsZWx4bWE=',
        path: 'code.py',
        text: 'Dangerous default value [] as argument',
        modifiedAt: '2020-07-06T10:14:07.226091+00:00',
        createdAt: '2020-07-06T10:14:07.226084+00:00',
        beginLine: 48,
        beginColumn: 17,
        endLine: 48,
        endColumn: 17,
        sourceCodeMarkup: '<div class="highlight"></div>',
        shortcode: 'PYL-W0102'
      },
      {
        id: 'Q2hlY2tJc3N1ZTp5dmxqZ2Q=',
        path: 'code.py',
        text: 'Dangerous default value {} as argument',
        modifiedAt: '2020-07-06T10:14:07.225798+00:00',
        createdAt: '2020-07-06T10:14:07.225791+00:00',
        beginLine: 29,
        beginColumn: 9,
        endLine: 29,
        endColumn: 9,
        sourceCodeMarkup: '<div class="highlight"></div>',
        shortcode: 'PYL-W0102'
      },
      {
        id: 'Q2hlY2tJc3N1ZTpta2xvcmc=',
        path: 'code.py',
        text: 'Dangerous default value [] as argument',
        modifiedAt: '2020-07-06T10:14:07.225495+00:00',
        createdAt: '2020-07-06T10:14:07.225488+00:00',
        beginLine: 23,
        beginColumn: 25,
        endLine: 23,
        endColumn: 25,
        sourceCodeMarkup: '<div class="highlight"></div>',
        shortcode: 'PYL-W0102'
      }
    ],
    issueIndex: 1,
    snippetsLoading: false,
    snippetsFetchErrored: false
  }

  const mocks = {
    $localStore: {
      get: () => ['']
    },
    $route: {
      params: {
        issueId: 'PYL-W0102'
      }
    }
  }

  const stubs = {
    IssueEditor: true,
    LazyEmptyState: true,
    ZButton: true,
    ZIcon: true,
    ZInput: true,
    ZMenu: true,
    ZMenuItem: true,
    ZPagination: true
  }

  test('renders a sample list of issue occurrences', () => {
    const { html } = render(IssueList, {
      mocks,
      propsData: baseProps,
      stubs
    })

    expect(html()).toMatchSnapshot()
  })
})
