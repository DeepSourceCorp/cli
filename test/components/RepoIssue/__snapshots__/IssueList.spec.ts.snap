// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ IssueList ]] renders a sample list of issue occurrences 1`] = `
"<div class=\\"flex flex-col space-y-4\\">
  <div class=\\"flex w-full flex-row-reverse gap-2 xl:flex-row\\">
    <zmenu-stub direction=\\"left\\" placement=\\"bottom\\" size=\\"base\\" width=\\"x-small\\" bodyspacing=\\"\\" triggerlabel=\\"Menu\\" itemszclass=\\"z-10\\" class=\\"text-vanilla-100 xl:hidden\\"></zmenu-stub>
    <zmenu-stub direction=\\"right\\" placement=\\"bottom\\" size=\\"base\\" width=\\"x-small\\" bodyspacing=\\"\\" triggerlabel=\\"Menu\\" itemszclass=\\"z-10\\" class=\\"hidden text-vanilla-100 xl:inline-block\\"></zmenu-stub>
    <div class=\\"h-full flex-grow xl:min-w-80\\">
      <zinput-stub name=\\"\\" value=\\"\\" label=\\"\\" type=\\"text\\" placeholder=\\"Search for issue title, file or issue code\\" debouncedelay=\\"350\\" size=\\"small\\" backgroundcolor=\\"ink-300\\" errorbordercolor=\\"cherry\\"></zinput-stub>
    </div>
  </div>
  <div class=\\"flex flex-col gap-y-3.5\\">
    <issueeditor-stub blob-url-root=\\"https://github.com/deepsourcelabs/demo-python/blob/2316e689facc8b4ea67258230b5cbce5c7d6c8d4/\\" can-ignore-issues=\\"true\\" check-id=\\"Q2hlY2s6enFkbmpv\\" check-issue-ids=\\"\\" shortcode=\\"PYL-W0102\\" id=\\"Q2hlY2tJc3N1ZTpsZWx4bWE=\\" path=\\"code.py\\" text=\\"Dangerous default value [] as argument\\" modifiedat=\\"2020-07-06T10:14:07.226091+00:00\\" createdat=\\"2020-07-06T10:14:07.226084+00:00\\" beginline=\\"48\\" begincolumn=\\"17\\" endline=\\"48\\" endcolumn=\\"17\\" sourcecodemarkup=\\"<div class=&quot;highlight&quot;></div>\\"></issueeditor-stub>
    <issueeditor-stub blob-url-root=\\"https://github.com/deepsourcelabs/demo-python/blob/2316e689facc8b4ea67258230b5cbce5c7d6c8d4/\\" can-ignore-issues=\\"true\\" check-id=\\"Q2hlY2s6enFkbmpv\\" check-issue-ids=\\"\\" shortcode=\\"PYL-W0102\\" id=\\"Q2hlY2tJc3N1ZTp5dmxqZ2Q=\\" path=\\"code.py\\" text=\\"Dangerous default value {} as argument\\" modifiedat=\\"2020-07-06T10:14:07.225798+00:00\\" createdat=\\"2020-07-06T10:14:07.225791+00:00\\" beginline=\\"29\\" begincolumn=\\"9\\" endline=\\"29\\" endcolumn=\\"9\\" sourcecodemarkup=\\"<div class=&quot;highlight&quot;></div>\\"></issueeditor-stub>
    <issueeditor-stub blob-url-root=\\"https://github.com/deepsourcelabs/demo-python/blob/2316e689facc8b4ea67258230b5cbce5c7d6c8d4/\\" can-ignore-issues=\\"true\\" check-id=\\"Q2hlY2s6enFkbmpv\\" check-issue-ids=\\"\\" shortcode=\\"PYL-W0102\\" id=\\"Q2hlY2tJc3N1ZTpta2xvcmc=\\" path=\\"code.py\\" text=\\"Dangerous default value [] as argument\\" modifiedat=\\"2020-07-06T10:14:07.225495+00:00\\" createdat=\\"2020-07-06T10:14:07.225488+00:00\\" beginline=\\"23\\" begincolumn=\\"25\\" endline=\\"23\\" endcolumn=\\"25\\" sourcecodemarkup=\\"<div class=&quot;highlight&quot;></div>\\"></issueeditor-stub>
    <!---->
  </div>
</div>"
`;
